R/add_Layer.R

```diff
@@ -1,27 +1,44 @@
-#' initializePlot
+#' initialize Plot
 #'
 #' Initialize a `ggplot` object
 #' with watermark
 #' and set its labels by metaData
 #'
-#' @param mapping  Default list of aesthetic mappings to use for plot
-#' @param metaData A named list of information about the `data` such as the `dimension` and `unit` of its variables.
+#' @param mappedData  MappedData object
+#' @param setMapping if TRUE (default) mapping is passed to ggplot, otherwise mapping will be used only to create labels
 #'
 #' @return A `ggplot` object
 #' @export
 #'
-initializePlot <- function(metaData = NULL,
-                           mapping = NULL) {
-  # Vlaidation
-  checkmate::assertList(metaData, types = "list", null.ok = TRUE)
-  checkmate::assertList(mapping, types = "quosure", null.ok = TRUE)
-  # initialize plot object
-  plotObject <- ggplot(environment = globalenv()) +
+initializePlot <- function(mappedData = NULL,
+                           setMapping = TRUE) {
+  # Validation
+  checkmate::assertClass(mappedData, classes = "MappedData", null.ok = TRUE)
```

Comment:
for many of this checks, **ospsuite.utils** provide validation functions, e.g. 
https://www.open-systems-pharmacology.org/OSPSuite.RUtils/reference/isOfType.html

Reply:
That was an intended omission
I want that the functions of this package are used and maybe copied and adjusted by people, with R experience but not so much ospsuite experience. So if there was a common known R function I have used that instead of an opsuite function.

Reply:
There are other reasons to use checkmate instead of the ospsuite.utils validation functions:

- Why write our own function, if there is already a package which does this perfectly
- with checkmate you can adress a list of validation topics (Type, any missing, kenght,..) in one line whereas you need in ospsuite utils one line per topic.
- Checkmate is much faster (see attached screenshot):
![Profvis_validation](https://github.com/Open-Systems-Pharmacology/OSPSuite.Plots/assets/28564329/038b9d6d-d414-4a08-8d73-a24d5ff8d0e1)

-